addons/hr_timesheet/controllers/portal.py

```diff
@@ -123,11 +122,15 @@ def get_timesheets():
             timesheets = Timesheet_sudo.search(domain, order=orderby, limit=_items_per_page, offset=pager['offset'])
             if field:
                 if groupby == 'date':
-                    time_data = Timesheet_sudo.read_group(domain, ['date', 'unit_amount:sum'], ['date:day'])
-                    mapped_time = dict([(datetime.strptime(m['date:day'], '%d %b %Y').date(), m['unit_amount']) for m in time_data])
-                    grouped_timesheets = [(Timesheet_sudo.concat(*g), mapped_time[k]) for k, g in groupbyelem(timesheets, itemgetter('date'))]
+                    raw_timesheets_group = Timesheet_sudo.read_group(
+                        domain, ["unit_amount:sum", "ids:array_agg(id)"], ["date:day"]
+                    )
+                    filtered_timesheets = [(timesheets.filtered(lambda t: t.id in d['ids']), d['unit_amount']) for d in raw_timesheets_group]
```

Comment:
Why do you perform a filtered? The domain passed to the read_group isn't it the same as the one provided to the search? If you need a recordset, a browse on the ids should be enough no? Or do I miss something?

Reply:
I used browse in the first place like this, but someone from the bugfix team suggested I use filtered instead.
I can revert it if you think it's better.

`grouped_timesheets = [
  (timesheets & Timesheet_sudo.browse(group["ids"]), group["unit_amount"])
  for group in grouped_timesheets
]`

Reply:
Filtered is a loop so you create a loop in a loop.
Using the browse prevents that, and should use the cache to get the values.
Don't for get to perform the browse in sudo then.